server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java

```diff
@@ -124,4 +127,44 @@ public String toString() {
             });
         }
     }
+
+    private static boolean matchesRequirement(DiscoveryNode discoveryNode, String requirement) {
+        return discoveryNode.getName().equals(requirement)
+            || discoveryNode.getAddress().toString().equals(requirement)
+            || discoveryNode.getAddress().getAddress().equals(requirement);
+    }
+
+    private static boolean checkWaitRequirements(GetDiscoveredNodesRequest request, Set<DiscoveryNode> nodes) {
+        List<String> requirements = request.getRequiredNodes();
+        if (requirements.size() != new HashSet<>(requirements).size()) {
+            throw new IllegalArgumentException("There are duplicate entries in [cluster.initial_master_nodes]");
```

Comment:
I think we don't need this, and if you omit it then you get more useful messages (i.e. it specifies one of the problematic requirements).

Reply:
Sure, removed it and adjusted test accordingly.